### PR TITLE
Fix TIC-80 when using a French AZERTY layout

### PR DESCRIPTION
**Pull request by aliceisjustplaying**
Fixes https://github.com/nesbox/TIC-80/issues/2520. `alt+[grave]12345` is not triggered if an AZERTY layout is set (the user can still use esc + function keys). This allows AZERTY keyboards to fully (as far as I can tell) work with TIC-80.

There is also some groundwork laid for a potential fix of https://github.com/nesbox/TIC-80/issues/1761 in the future, as well as detecting other layouts; this is more work though and definitely a separate PR. Let me know if you'd rather not have this bit and I can dumb it down to strictly detecting AZERTY.

Tested on MacOS 14, Windows 11 and Linux (EndeavourOS/Arch + Wayland + KDE Plasma 6).

Notes for Linux:
* All SDL2 apps, therefore TIC-80 as well run under Xwayland by default. Keyboard layout change detection will only work if done via `setxkbmap` (or something that calls that). If I change the keyboard from KDE Plasma 6 directly, the layout change itself it is not passed down/detected. There isn't anything we can do about this.
* TIC-80 doesn't work when forced to Wayland (`SDL_VIDEODRIVER=wayland`), only Xwayland. It appears to start and has audio but no window is shown. That's a separate bug to track down. Created #2660.
* alt+1 will insert a `1` for some reason when already on the code tab.  That's a separate bug to track down. Created #2661.

### CODE REVIEW

File: src/system/sdl/main.c [hunk 1]
```diff
@@ -1062,6 +1062,26 @@ static void handleKeydown(SDL_Keycode keycode, bool down, bool* state, bool* pre
...
 
+char* detect_keyboard_layout() 
+{
+    char q = SDL_GetKeyFromScancode(SDL_SCANCODE_Q);
+    char w = SDL_GetKeyFromScancode(SDL_SCANCODE_W);
+    char y = SDL_GetKeyFromScancode(SDL_SCANCODE_Y);
+
+    char* layout = "unknown";
+
+    if (q == 'q' && w == 'w' && y == 'y') layout = "qwerty"; // US etc.
```
Comment (nesbox) [inline src/system/sdl/main.c:1073]: can we use enum here instead of strings?
smth like
```c
typedef enum
{
tic_layout_qwerty,
...
} tic_layout;
```

**Review by nesbox — CHANGES_REQUESTED** [review]

**Comment by nesbox** [comment]
@aliceisjustplaying now we have a conflict, could you pls resolve it?\
Thanks

File: src/system/sdl/main.c (hunk 1 above)
Comment (aliceisjustplaying) [reply src/system/sdl/main.c]: Done in 4f54237

**Comment by aliceisjustplaying** [comment]
Merge conflict fixed

**Review by nesbox — APPROVED** [review]
